docs/Inspection-Concept.md

```diff
@@ -0,0 +1,80 @@
+# Machinery inspection concept
+
+This document describes the concept of inspecting systems in Machinery. System
+inspection is one of the central features of Machinery. It gathers all data
+about a given system and stores it in a description. The description can then be
+used for a variety of use cases, including display, archive, export to other
+formats, and more. Inspection is implemented as the `machinery inspect` command.
+
+## Access to inspected system
+
+To gather data Machinery executes a number of commands on the inspected system,
+collects and processes the results, and stores them in the format of a system
+description on the local system, where Machinery is executed.
+
+Usually the inspected system will be different from the system where Machinery
+is running. Machinery will use ssh to access the remote system. For the special
+case where the inspected system is the local system where Machinery is run, the
+same commands will be executed but not via ssh but locally.
```

Comment:
I'd remove one "but" from the last sentence
`... will be executed not via ssh but locally `